lib/builtin.t

```diff
@@ -176,4 +176,79 @@ package FetchStoreCounter {
     cmp_ok($val, $_,  !1, "false is equivalent to  !1 by $_") for qw( eq == );
 }
 
+# Vanilla trim tests
+{
+    use builtin qw( trim );
+
+    is(trim("    Hello world!   ")      , "Hello world!"  , 'Trim spaces');
+    is(trim("\tHello world!\t")         , "Hello world!"  , 'Trim tabs');
+    is(trim("\n\n\nHello\nworld!\n")    , "Hello\nworld!" , 'Trim \n');
+    is(trim("\t\n\n\nHello world!\n \t"), "Hello world!"  , 'Trim all three');
+    is(trim("Perl")                     , "Perl"          , 'Trim nothing');
+    is(trim('')                         , ""              , 'Trim empty string');
+}
+
+TODO: {
+    my $warn = '';
+    local $SIG{__WARN__} = sub { $warn .= join "", @_; };
+
+    is(builtin::trim(undef), ""                                        , 'Trim undef');
+    like($warn    , qr/^Use of uninitialized value in subroutine entry at/, 'Trim undef triggers warning');
```

Comment:
Yeah these are quite annoying. I want to fix that in a nice way but I don't think there's time to get that into 5.36 now, so this will have to do. We'll find a better way in 5.37.